src/aap_eda/wsapi/consumers.py

```diff
@@ -513,3 +512,43 @@ def get_file_contents_from_credentials(
                     contents = substitute_variables(value, inputs)
                     file_templates[template] = contents
         return file_templates
+
+    @database_sync_to_async
+    def get_env_vars_from_credentials(
+        self, activation: models.Activation
+    ) -> tp.Optional[str]:
+        env_vars = {}
+        vault_inputs = {}
+        if activation.eda_system_vault_credential:
+            vault_inputs = activation.eda_system_vault_credential.inputs
+            vault_inputs = yaml.safe_load(vault_inputs.get_secret_value())
+
+        for eda_credential in activation.eda_credentials.all():
+            schema_inputs = eda_credential.credential_type.inputs
+            secret_fields = get_secret_fields(schema_inputs)
+            injectors = eda_credential.credential_type.injectors
+            user_inputs = yaml.safe_load(
+                eda_credential.inputs.get_secret_value()
+            )
+            if "env" in injectors:
+                for key, value in user_inputs.items():
+                    if key in secret_fields:
+                        user_inputs[key] = encrypt_string(
+                            password=vault_inputs["vault_password"],
+                            plaintext=value,
+                            vault_id=vault_inputs["vault_id"],
+                        )
+
+                for key, value in injectors["env"].items():
+                    if key in env_vars:
+                        raise DuplicateEnvKeyError(f"env {key} already exists")
+                    env_vars[key] = (
+                        value
+                        if not isinstance(value, str)
+                        or "eda.filename" in value
```

Comment:
Why do we need to handle this especially? it mean the value with `eda.filename` always doesn't need to be substituted?

Reply:
@hsong-rh Its only available after the file is created by ansible-rulebook if you do the substitution on the server side that will fail. So it has to be bypassed.